<a href="https://colab.research.google.com/github/kavithachitriki/SCALER_PRACTICE/blob/main/Netflix_Scaler.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [54]:
import pandas as pd
import warnings
warnings.filterwarnings("ignore")
import io
import seaborn as sns
from google.colab import files


In [55]:
#uploaded=files.upload()

In [56]:
!gdown 1-LPoouCG4CGN2h5dOxQithXYvJQAf389

Downloading...
From: https://drive.google.com/uc?id=1-LPoouCG4CGN2h5dOxQithXYvJQAf389
To: /content/netflix.csv
100% 3.40M/3.40M [00:00<00:00, 227MB/s]


In [57]:
df=pd.read_csv("netflix.csv")

In [58]:
df.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

In [59]:
df.shape

(8807, 12)

It is observed that Data is having 8807 rows and 12 features


In [60]:
df.isnull().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

It is observed that director column is having more null values, and cast and country also has more null values


In [61]:
df["director"].fillna("Unknown",inplace=True)
df["cast"].fillna("Unknown",inplace=True)
df["country"].fillna("Unknown",inplace=True)

We are replacing NaN values with "Unknown" in director,cast and country Column so as not to lose data


In [62]:
df["duration"].fillna(df["duration"].mode(),inplace=True)
df["date_added"].fillna(df["date_added"].mode(),inplace=True)


In [63]:
df["type"].value_counts(ascending=True)

TV Show    2676
Movie      6131
Name: type, dtype: int64

We have 2676 TVShows and 6131 Movies in data,Movies has high frequency



In [64]:
constraint=df["cast"].apply(lambda x: str(x).split(',')).tolist()

In [65]:
constraint[0],constraint[1],constraint[2],

(['Unknown'],
 ['Ama Qamata',
  ' Khosi Ngema',
  ' Gail Mabalane',
  ' Thabang Molaba',
  ' Dillon Windvogel',
  ' Natasha Thahane',
  ' Arno Greeff',
  ' Xolile Tshabalala',
  ' Getmore Sithole',
  ' Cindy Mahlangu',
  ' Ryle De Morny',
  ' Greteli Fincham',
  ' Sello Maake Ka-Ncube',
  ' Odwa Gwanya',
  ' Mekaila Mathys',
  ' Sandi Schultz',
  ' Duane Williams',
  ' Shamilla Miller',
  ' Patrick Mofokeng'],
 ['Sami Bouajila',
  ' Tracy Gotoas',
  ' Samuel Jouy',
  ' Nabiha Akkari',
  ' Sofia Lesaffre',
  ' Salim Kechiouche',
  ' Noureddine Farihi',
  ' Geert Van Rampelberg',
  ' Bakary Diombera'])

In [66]:
constraint[5]

['Kate Siegel',
 ' Zach Gilford',
 ' Hamish Linklater',
 ' Henry Thomas',
 ' Kristin Lehman',
 ' Samantha Sloyan',
 ' Igby Rigney',
 ' Rahul Kohli',
 ' Annarah Cymone',
 ' Annabeth Gish',
 ' Alex Essoe',
 ' Rahul Abburi',
 ' Matt Biedel',
 ' Michael Trucco',
 ' Crystal Balint',
 ' Louis Oliver']

In [67]:
df_new=pd.DataFrame(constraint,index=df["title"])

In [68]:
df_new.head()

0                1               2   \
title                                                                   
Dick Johnson Is Dead         Unknown             None            None   
Blood & Water             Ama Qamata      Khosi Ngema   Gail Mabalane   
Ganglands              Sami Bouajila     Tracy Gotoas     Samuel Jouy   
Jailbirds New Orleans        Unknown             None            None   
Kota Factory              Mayur More   Jitendra Kumar      Ranjan Raj   

                                    3                  4                  5   \
title                                                                          
Dick Johnson Is Dead              None               None               None   
Blood & Water           Thabang Molaba   Dillon Windvogel    Natasha Thahane   
Ganglands                Nabiha Akkari     Sofia Lesaffre   Salim Kechiouche   
Jailbirds New Orleans             None               None               None   
Kota Factory                 Alam Khan      Ahsaas Channa     Revathi Pillai   

                                       6                      7   \
title                                                              
Dick Johnson Is Dead                 None                   None   
Blood & Water                 Arno Greeff      Xolile Tshabalala   
Ganglands               Noureddine Farihi   Geert Van Rampelberg   
Jailbirds New Orleans                None                   None   
Kota Factory                   Urvi Singh             Arun Kumar   

                                     8                9   ...    40    41  \
title                                                     ...               
Dick Johnson Is Dead               None             None  ...  None  None   
Blood & Water           Getmore Sithole   Cindy Mahlangu  ...  None  None   
Ganglands               Bakary Diombera             None  ...  None  None   
Jailbirds New Orleans              None             None  ...  None  None   
Kota Factory                       None             None  ...  None  None   

                         42    43    44    45    46    47    48    49  
title                                                                  
Dick Johnson Is Dead   None  None  None  None  None  None  None  None  
Blood & Water          None  None  None  None  None  None  None  None  
Ganglands              None  None  None  None  None  None  None  None  
Jailbirds New Orleans  None  None  None  None  None  None  None  None  
Kota Factory           None  None  None  None  None  None  None  None  

[5 rows x 50 columns]

In [69]:
df_new=df_new.stack()
df_new.head(40)

title                    
Dick Johnson Is Dead   0                   Unknown
Blood & Water          0                Ama Qamata
                       1               Khosi Ngema
                       2             Gail Mabalane
                       3            Thabang Molaba
                       4          Dillon Windvogel
                       5           Natasha Thahane
                       6               Arno Greeff
                       7         Xolile Tshabalala
                       8           Getmore Sithole
                       9            Cindy Mahlangu
                       10            Ryle De Morny
                       11          Greteli Fincham
                       12     Sello Maake Ka-Ncube
                       13              Odwa Gwanya
                       14           Mekaila Mathys
                       15            Sandi Schultz
                       16           Duane Williams
                       17          Shamilla Miller
     

In [70]:
df_new=pd.DataFrame(df_new.reset_index())

In [71]:
df_new.head()

title  level_1                0
0  Dick Johnson Is Dead        0          Unknown
1         Blood & Water        0       Ama Qamata
2         Blood & Water        1      Khosi Ngema
3         Blood & Water        2    Gail Mabalane
4         Blood & Water        3   Thabang Molaba

In [72]:
df_new.rename(columns={0:'Actors'},inplace=True)
df_new.drop(["level_1"],axis=1,inplace=True)
df_new.head()

title           Actors
0  Dick Johnson Is Dead          Unknown
1         Blood & Water       Ama Qamata
2         Blood & Water      Khosi Ngema
3         Blood & Water    Gail Mabalane
4         Blood & Water   Thabang Molaba

In [74]:
df_final=df_new.merge(df[["show_id","type","title","director","country","date_added","release_year","rating","duration","listed_in","description"]],on=["title"],how="left")

In [75]:
df_final.head()

title           Actors show_id     type         director  \
0  Dick Johnson Is Dead          Unknown      s1    Movie  Kirsten Johnson   
1         Blood & Water       Ama Qamata      s2  TV Show          Unknown   
2         Blood & Water      Khosi Ngema      s2  TV Show          Unknown   
3         Blood & Water    Gail Mabalane      s2  TV Show          Unknown   
4         Blood & Water   Thabang Molaba      s2  TV Show          Unknown   

         country          date_added  release_year rating   duration  \
0  United States  September 25, 2021          2020  PG-13     90 min   
1   South Africa  September 24, 2021          2021  TV-MA  2 Seasons   
2   South Africa  September 24, 2021          2021  TV-MA  2 Seasons   
3   South Africa  September 24, 2021          2021  TV-MA  2 Seasons   
4   South Africa  September 24, 2021          2021  TV-MA  2 Seasons   

                                         listed_in  \
0                                    Documentaries   
1  International TV Shows, TV Dramas, TV Mysteries   
2  International TV Shows, TV Dramas, TV Mysteries   
3  International TV Shows, TV Dramas, TV Mysteries   
4  International TV Shows, TV Dramas, TV Mysteries   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  After crossing paths at a party, a Cape Town t...  
3  After crossing paths at a party, a Cape Town t...  
4  After crossing paths at a party, a Cape Town t...

In [76]:
df_final.shape

(64951, 12)

In [77]:
df["rating"].value_counts()

TV-MA       3207
TV-14       2160
TV-PG        863
R            799
PG-13        490
TV-Y7        334
TV-Y         307
PG           287
TV-G         220
NR            80
G             41
TV-Y7-FV       6
NC-17          3
UR             3
74 min         1
84 min         1
66 min         1
Name: rating, dtype: int64

In [79]:
import matplotlib.pyplot as plt
df_actors=df_final.groupby(["Actors"]).agg({"title":"nunique"}).reset_index()
df_actors["Actors"].replace(["nan"],["Unknown Actor"],inplace=True)
df_actors=df_actors[df_actors["Actors"] != "Unknown Actor"]
plt.figure(figsize=(15,8))
plt.barh(df_actors[::-1]["Actors"],df_actors[::-1]["title"],color=["red"])
plt.xlabel("Number of Movies")
plt.ylabel("Popular Actors")
plt.show()

ValueError: ignored

<Figure size 1080x576 with 1 Axes>